moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -211,30 +211,52 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     }
   }
 
-  void waitForExecuteActionOrService(ros::WallTime timeout_for_servers)
+  void waitForExecuteActionOrService(ros::WallTime timeout)
   {
+    ROS_DEBUG("Waiting for move_group action server (%s)...", move_group::EXECUTE_ACTION_NAME.c_str());
+
+    // Deprecated service
     execute_service_ =
         node_handle_.serviceClient<moveit_msgs::ExecuteKnownTrajectory>(move_group::EXECUTE_SERVICE_NAME);
 
     // wait for either of action or service
-    while (!execute_action_client_->isServerConnected() && !execute_service_.exists() &&
-           timeout_for_servers > ros::WallTime::now())
+    if (timeout == ros::WallTime()) // wait forever
+    {
+      while (!execute_action_client_->isServerConnected() && !execute_service_.exists())
+      {
+        ros::WallDuration(0.001).sleep();
+        // explicit ros::spinOnce on the callback queue used by NodeHandle that manages the action client
+        ((ros::CallbackQueue *)node_handle_.getCallbackQueue())->callAvailable();
+      }
+    }
+    else // wait with timeout
     {
-      ros::WallDuration(0.001).sleep();
-      // explicit ros::spinOnce on the callback queue used by NodeHandle that manages the action client
-      ((ros::CallbackQueue *)node_handle_.getCallbackQueue())->callAvailable();
+      while (!execute_action_client_->isServerConnected() && !execute_service_.exists() &&
+             timeout > ros::WallTime::now())
+      {
+        ros::WallDuration(0.001).sleep();
+        // explicit ros::spinOnce on the callback queue used by NodeHandle that manages the action client
+        ((ros::CallbackQueue *)node_handle_.getCallbackQueue())->callAvailable();
+      }
     }
 
     // issue warning
     if (!execute_action_client_->isServerConnected())
     {
       if (execute_service_.exists())
+      {
         ROS_WARN_NAMED("planning_interface",
                        "\nDeprecation warning: Trajectory execution service is deprecated (was replaced by an action)."
                        "\nReplace 'MoveGroupExecuteService' with 'MoveGroupExecuteTrajectoryAction' in "
                        "move_group.launch");
+      }
       else
+      {
+        ROS_ERROR_STREAM_NAMED("planning_interface", "Unable to find execution action on topic: "
+                               << node_handle_.getNamespace() + move_group::EXECUTE_ACTION_NAME
+                               << " or service: " << node_handle_.getNamespace() + move_group::EXECUTE_SERVICE_NAME);
         throw std::runtime_error("No Trajectory execution capability available.");
```

Comment:
This was previously always being thrown when default MoveGroupInterface constructor used